code/src/components/AccessPage.spec.tsx

```diff
@@ -0,0 +1,44 @@
+import '@testing-library/jest-dom';
+import { render, screen } from '@testing-library/react';
+import { fireEvent } from '@testing-library/dom';
+import AccessPage from './AccessPage';
+
+jest.mock('next/navigation', () => ({
+  useRouter() {
+    return {};
+  },
+}));
+
+describe('SignInForm', () => {
+  test('renders SignInForm', () => {
+    const { getByPlaceholderText } = render(<AccessPage />);
+
+    expect(getByPlaceholderText('Email')).toBeInTheDocument();
+    expect(getByPlaceholderText('Password')).toBeInTheDocument();
+  });
+
+  test('redirect to SignUp page', () => {
+    const { getByText, getByTestId } = render(<AccessPage />);
+
+    expect(getByTestId('to-signup')).toBeTruthy();
+
+    fireEvent.click(getByTestId('to-signup'));
+
+    expect(getByText('Sign Up a New Account')).toBeInTheDocument();
```

Comment:
It's kind of bad practise to select by text anywhere because it doesn't work for multilanguage websites and cms and you have to fix test every time you change text

Reply:
Good point, but I don't think the scope of this project is big enough